snuba/datasets/errors_replacer.py

```diff
@@ -667,15 +665,92 @@ def process_unmerge(
         "previous_group_id": message["previous_group_id"],
         "project_id": message["project_id"],
         "timestamp": timestamp.strftime(DATETIME_FORMAT),
+        "hashes": ", ".join(_convert_hash(h, state_name) for h in hashes),
     }
 
+    query_time_flags = (NEEDS_FINAL, message["project_id"])
+
+    return LegacyReplacement(
+        count_query_template, insert_query_template, query_args, query_time_flags
+    )
+
+
+def _convert_hash(
+    hash: str, state_name: ReplacerState, convert_types: bool = False
+) -> str:
     if state_name == ReplacerState.ERRORS:
-        query_args["hashes"] = ", ".join(
-            ["'%s'" % str(uuid.UUID(_hashify(h))) for h in hashes]
-        )
+        if convert_types:
+            return "toUUID('%s')" % str(uuid.UUID(_hashify(hash)))
+        else:
+            return "'%s'" % str(uuid.UUID(_hashify(hash)))
     else:
-        query_args["hashes"] = ", ".join("'%s'" % _hashify(h) for h in hashes)
+        if convert_types:
+            return "toFixedString('%s', 32)" % _hashify(hash)
+        else:
+            return "'%s'" % _hashify(hash)
+
+
+def process_unmerge_hierarchical(
+    message: Mapping[str, Any],
+    all_columns: Sequence[FlattenedColumn],
+    state_name: ReplacerState,
+) -> Optional[Replacement]:
+    timestamp = datetime.strptime(message["datetime"], settings.PAYLOAD_DATETIME_FORMAT)
+    all_column_names = [c.escaped for c in all_columns]
+    select_columns = map(
+        lambda i: i if i != "group_id" else str(message["new_group_id"]),
+        all_column_names,
+    )
+
+    primary_hash = message["primary_hash"]
+    assert isinstance(primary_hash, str)
+
+    hierarchical_hash = message["hierarchical_hash"]
+    assert isinstance(hierarchical_hash, str)
```

Comment:
For sure. I also believe this is a replacement that could be DLQ'd and re-inserted safely, while it would hang in the DLQ one would have data inconsistencies but not beyond that.